lib/methods/ifwapichecktask.cpp

```diff
@@ -344,6 +251,30 @@ void IfwApiCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRes
 	String username = resolveMacros("$ifw_api_username$");
 	String password = resolveMacros("$ifw_api_password$");
 
+	std::function<void()> resultHandler;
```

Comment:
I don't think that `ReportIfwCheckResult` was a bad function name, something like `reportResult` actually describes pretty well that it does. Maybe add a little comment that the rest of the code is supposed to fill `cr` and then call it, which submits it (as is has captured it, that's mostly why I initially suggested to have `cr` as a parameter: `cr->Set...(); reportResult(cr);` is just more obvious than `cr->Set...(); reportResult();` + "oh by the way, that's a lambda that has captures the very same `cr` some time ago").